components/TokenManagePage/Media.tsx

```diff
@@ -19,6 +33,28 @@ const Media = () => {
     if (!description) setDescription(meta.description || "");
   }, [meta, title, description]);
 
+  const isOwner = Boolean(
+    connectedAddress && owner &&
+    getAddress(connectedAddress) === getAddress(owner)
+  );
+
+  const onSave = async () => {
+    if (!isOwner || !token || !collection) return;
+    try {
+      const newUri = await generateNewMetadataFromToken(
+        collection.chainId,
+        token.tokenContractAddress,
+        token.tokenId,
+        title,
+        description,
+      );
+      await setMetadata(newUri);
+      toast.success("Metadata updated successfully!");
+    } catch (e) {
+      toast.error("Failed to save metadata");
+    }
+  };
```

Comment:
Single Responsibility Principle 
- Actual: defining function logic within a react component file
- Required: new `lib` or `hook` for the `onSave` function which is then imported into the `<Media />` component.